### PR TITLE
feat: Add governance calls

### LINKED ISSUE #56: Add governance calls
Add the following dispatchable functions from `democracy`:

- `set_proxy`
- `resign_proxy`
- `remove_proxy`
- `proxy_vote`
- `vote`

### PR DESCRIPTION
**Pull request by axelchalon**
close #56

### CODE REVIEW

File: src/methods/democracy/types.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/**
+ * A vote in a referendum
+ */
+export type Vote = {
```
Comment (amaury1093) [inline src/methods/democracy/types.ts:4]: Since this Vote is really specific to proxyVote, I'd rather have this definition inside proxyVote.ts

Comment (axelchalon) [reply src/methods/democracy/types.ts:4]: `Vote` is used in vote.ts and proxyVote.ts

Comment (amaury1093) [reply src/methods/democracy/types.ts:4]: Ah, right! Then everything's good

**Review by amaury1093 — COMMENTED** [review]
Nice! lgtm for now. Could you:
- Add the decoding tests in the `decode*.spec.ts` for these methods/
- Run `yarn docs`?

**Comment by amaury1093** [comment]
ah, the errors come from the fact that you need to add a couple of index.ts, and export them: https://github.com/paritytech/txwrapper/blob/master/src/methods/index.ts#L3

**Comment by axelchalon** [comment]
Hmm, tests are failing as txInfo.method.args.vote is encoded into bytes vs expected json object. The Call method for vote/proxyVote has the argument `vote` with the type `[Uint8Array]` which doesn't seem right. However, I don't know how I can specify to encode it as a struct instead. Any pointers?

**Comment by amaury1093** [comment]
The cleanest would be imo to add a correct `.toJSON()` here: https://github.com/polkadot-js/api/blob/b7eeb992cd1f05d77bb5371a1ffa3894f968f042/packages/types/src/primitive/Generic/Vote.ts

But i'm okay to skip those tests for now.

**Comment by axelchalon** [comment]
> The cleanest would be imo to add a correct `.toJSON()` here: https://github.com/polkadot-js/api/blob/b7eeb992cd1f05d77bb5371a1ffa3894f968f042/packages/types/src/primitive/Generic/Vote.ts
> 
> But i'm okay to skip those tests for now.

Hmm, `Vote` extends `U8aFixed` which extends `Raw` whose `toJSON` signature requires the return value to be a `string`, so we cannot make the `toJSON` of `Vote` be an object. I imagine a solution would be to make `Vote` extend Struct? although I don't know if it is encoded that way

**Review by amaury1093 — APPROVED** [review]
lgtm! I'm okay to merge this

**Comment by amaury1093** [comment]
> Hmm, Vote extends U8aFixed which extends Raw whose toJSON signature requires the return value to be a string, so we cannot make the toJSON of Vote be an object. I imagine a solution would be to make Vote extend Struct? although I don't know if it is encoded that way

Ah yeah, true, Vote extend Struct could work too. Though tbh I don't think it's actually worth the effort, you'd need to overwrite Struct's encoding/decoding.

I'm really okay to leave Vote as-is, showing a hex. If _really_ we wanted to show a JSON object, we could add a special check here: https://github.com/paritytech/txwrapper/blob/02978cb68746b623aa839492e10f4ec805f8c543/src/util/method.ts#L113

something like if codec is a Vote, then don't use toJSON, but some special function.

**Pull request merged**